File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        curr_count = 1
        curr_number = -1

        for number in nums:
            if number != curr_number:
                curr_count -= 1
                if curr_count == 0:
                    curr_count = 1
                    curr_number = number
            else:
                curr_count += 1
        
        return curr_number

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let dictionary = {}

    nums.forEach((curr)=>{
        if(curr in dictionary){
            dictionary[curr]+=1
        }else{
            dictionary[curr] = 1
        }
    })

    for(let key in dictionary){
        if(dictionary[key] > Math.floor(nums.length/2)){
            return key
        }
    }
};